File: geometry/cpp/no1085.cpp
#include <iostream>
#include <deque>
#include <algorithm>

using namespace std;

int main() {
    int x, y, w, h;
    cin >> x >> y >> w >> h;

    int x_to_w = w - x;
    int y_to_h = h - y;

    int x_to_zero = x;
    int y_to_zero = y;

    deque<int> results;
    results.push_back(x_to_w);
    results.push_back(y_to_h);
    results.push_back(x_to_zero);
    results.push_back(y_to_zero);

    cout << *min_element(results.begin(), results.end()) << endl; 
    
    return 0;
}

File: geometry/python/no1085.py
x, y, w, h = map(int, input().split())

y_to_h = h - y
x_to_w = w - x

x_to_zero = x - 0
y_to_zero = y - 0

results = [y_to_h, x_to_w, x_to_zero, y_to_zero]
print(min(results))